gensim/models/ldamulticore.py

```diff
@@ -289,7 +289,7 @@ def process_result_queue(force=False):
                 # put the chunk into the workers' input job queue
                 while True:
                     try:
-                        job_queue.put((chunk_no, chunk, self), block=False)
+                        job_queue.put((chunk_no, chunk, self.state), block=False)
```

Comment:
I think this is a little bit confusing.

The pool workers receive `self` as a parameter. Why are we passing `self.state` via the job queue as well?

Reply:
I guess the reason is simple: `RAM_USAGE(self) > RAM_USAGE(self.state)`

Reply:
But why do we need access to self.state when we already have self?

Reply:
good question, idk
@horpto ?

Reply:
@horpto Ping.

Reply:
@horpto Can you please comment on this?

When we're [setting up the pool](https://github.com/RaRe-Technologies/gensim/pull/2300/files#diff-68941d8c838f8d70a6648f06b9866889R280), we're calling it like this:

```python
pool = Pool(self.workers, worker_e_step, (job_queue, result_queue, self))
```

The first arg is the number of workers, the second is an initializer function, and the third is a set of parameters to pass to the initializer ([docs](https://docs.python.org/3.8/library/multiprocessing.html#multiprocessing.pool.Pool)). The workers will call this initializer with the args on start-up. 

> If initializer is not None then each worker process will call initializer(*initargs) when it starts.

So, as you can see, we are already passing `self` to the workers.

Why do we need to pass `self` again here?

Reply:
> Why do we need to pass self again here?

Well, it's `multiprocessing` and synchronization issue.  Worker and master processes will have different state. When `Pool`  is called it will create `self.workers` processes (via [_repopulate_pool_static method](https://github.com/python/cpython/blob/8bea91b5e9ea07ca93958e131b436024f0b1b1cf/Lib/multiprocessing/pool.py#L312)) with our initargs. Current process class creates system process (2-3 different ways and especial pain is win) and then sends data to children process with pickle and pipe (for example [spawn_win32](https://github.com/python/cpython/blob/8bea91b5e9ea07ca93958e131b436024f0b1b1cf/Lib/multiprocessing/popen_spawn_win32.py#L93).) Of course, there is a fork type process [(fork popen)](https://github.com/python/cpython/blob/master/Lib/multiprocessing/popen_fork.py) but it's not crossplatform. What's more [multiprocess.Queue](https://github.com/python/cpython/blob/master/Lib/multiprocessing/queues.py#L35) also uses pipe+pickle so has overhead.